rust/lance-encoding/src/data.rs

```diff
@@ -95,15 +157,30 @@ impl NullableDataBlock {
         Self {
             data: Box::new(self.data.borrow_and_clone()),
             nulls: self.nulls.borrow_and_clone(),
+            block_info: self.block_info.clone(),
+            used_encoding: self.used_encoding.clone(),
         }
     }
 
     fn try_clone(&self) -> Result<Self> {
         Ok(Self {
             data: Box::new(self.data.try_clone()?),
             nulls: self.nulls.try_clone()?,
+            block_info: self.block_info.clone(),
+            used_encoding: self.used_encoding.clone(),
         })
     }
+
+    // count_nulls will be handled differently after V2.1
+    pub fn count_nulls(&mut self) -> u64 {
+        let nulls_buf = &self.borrow_and_clone().into_buffers()[0];
```

Comment:
Can you access `self.nulls` directly (e.g. `let nulls_buf = self.nulls.borrow_and_clone()`)?  I don't think you need `into_buffers`?

Reply:
sorry, the `count_nulls` method here is vestige, I plan to remove `count_nulls` for now as `NullableDataBlock` will be removed soon